libs/service-portal/settings/personal-information/src/screens/EditNudge/EditNudge.tsx

```diff
@@ -19,39 +22,30 @@ import {
 import React, { useState } from 'react'
 import { NudgeForm, NudgeFormData } from '../../components/Forms/NudgeForm'
 
-const UpdateIslykillSettings = gql`
-  mutation updateIslykillSettings($input: UpdateIslykillSettingsInput!) {
-    updateIslykillSettings(input: $input) {
-      nationalId
-    }
-  }
-`
-
 export const EditNudge: ServicePortalModuleComponent = () => {
   useNamespaces('sp.settings')
-  const { data: settings } = useUserProfileAndIslykill()
+  const { data: settings } = useUserProfile()
   const [status, setStatus] = useState<'passive' | 'success' | 'error'>(
     'passive',
   )
+  const { createUserProfile } = useCreateUserProfile()
+  const { updateUserProfile } = useUpdateUserProfile()
 
-  const [updateIslykill, { loading, error }] = useMutation(
-    UpdateIslykillSettings,
-  )
   const { formatMessage } = useLocale()
 
   const submitFormData = async (formData: NudgeFormData) => {
     if (status !== 'passive') setStatus('passive')
 
     try {
-      await updateIslykill({
-        variables: {
-          input: {
-            email: settings?.email,
-            mobile: settings?.mobile,
-            canNudge: formData.nudge,
-          },
-        },
-      })
+      if (settings) {
+        await updateUserProfile({
+          canNudge: formData.nudge,
```

Comment:
Does the UserProfile and Islykill APIs support partial updates? We wouldn’t want to accidentally clear saved locale/email/phone/bankinfo.

In standard REST APIs, PUT requests don’t support partial updates, and require all fields to be included (compared with PATCH).